packages/compass-aggregations/src/components/pipeline-explain/explain-query-performance.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { Body, Subtitle, css, spacing } from '@mongodb-js/compass-components';
+import type { IndexInformation } from '@mongodb-js/explain-plan-helper';
+
+import { ExplainIndexes } from './explain-indexes';
+
+type ExplainQueryPerformanceProps = {
+  executionTimeMillis: number;
+  nReturned: number;
+  usedIndexes: IndexInformation[];
+};
+
+const containerStyles = css({
+  display: 'flex',
+  gap: spacing[3],
+  flexDirection: 'column',
+});
+
+const statsStyles = css({
+  gap: spacing[1],
+  display: 'flex',
+  flexDirection: 'column',
+});
+
+const statItemStyles = css({
+  display: 'flex',
+  gap: spacing[1],
+});
+
+const statTitleStyles = css({
+  whiteSpace: 'nowrap',
+});
+
+export const ExplainQueryPerformance: React.FunctionComponent<ExplainQueryPerformanceProps> =
+  ({ nReturned, executionTimeMillis, usedIndexes }) => {
+    return (
+      <div
+        className={containerStyles}
+        data-testid="pipeline-explain-results-summary"
+      >
+        <Subtitle>Query Performance Summary</Subtitle>
+        <div className={statsStyles}>
+          {nReturned && (
```

Comment:
Zero is falsy, so if we want zeroes to show up, you probably want something like this instead

```suggestion
          {typeof nReturned === 'number' && (
```